src/Generation/BuildMethodFragmentGenerator.php

```diff
@@ -0,0 +1,214 @@
+<?php
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+declare(strict_types=1);
+
+namespace Google\Generator\Generation;
+
+use Google\Generator\Ast\AST;
+use Google\Generator\Ast\Access;
+use Google\Generator\Ast\PhpClassMember;
+use Google\Generator\Ast\PhpDoc;
+use Google\Generator\Ast\PhpParam;
+use Google\Generator\Collections\Map;
+use Google\Generator\Collections\Vector;
+use Google\Generator\Utils\Formatter;
+use Google\Generator\Utils\Helpers;
+use Google\Generator\Utils\ResolvedType;
+use Google\Generator\Utils\Type;
+
+class BuildMethodFragmentGenerator
+{
+    public static function generate(SourceFileContext $ctx, ServiceDetails $serviceDetails): Map
+    {
+        return (new BuildMethodFragmentGenerator($ctx, $serviceDetails))->generateImpl();
+    }
+
+    public static function format(string $code): string
+    {
+        // Add a class wrapper so that the formatter works (requires valid PHP)
+        $codeWrap = "<?php\n\nnew class {\n%s\n};";
+
+        // Apply the standard Formatter
+        $code = Formatter::format(sprintf($codeWrap, $code));
+
+        // Remove the wrapping class
+        preg_match(sprintf('/' . $codeWrap . '/s', '(.*)'), $code, $matches);
+
+        return $matches[1] . PHP_EOL;
+    }
+
+    private function generateImpl(): Map
+    {
+        $buildMethodSnippets = Map::new([]);
+        foreach ($this->serviceDetails->methods as $methodDetails) {
+            if ($methodDetails->methodSignature) {
+                $buildMethods = Vector::new();
+                foreach ($methodDetails->methodSignature as $i => $methodSignature) {
+                    if (empty($methodSignature)) {
+                        continue;
+                    }
+                    $buildMethods = $buildMethods->append(
+                        $this->buildMethodSnippet($methodDetails, $methodSignature, 0 === $i)
+                    );
+                }
+                if ($buildMethods->count() > 0) {
+                    $buildMethodSnippets = $buildMethodSnippets->set(
+                        $methodDetails->requestType->name,
```

Comment:
```suggestion
                        str_replace('\\', '/', $methodDetails->requestType->getFullname(true)),
```

So the fragements are `Google/Cloud/Foo/V1/GetFooRequest.build.txt`